### ISSUE TITLE
SetBucketLifecycle sends no content to server

### ISSUE THREAD

**Issue by hoempf** [minio/minio-go#1205]
### Bug Report

I've just tried to set a content lifecycle configuration for my bucket and got 204 replies from the server (no content).

### How to reproduce

Version: v6.0.44 (latest at the time)
Follow the example here: https://github.com/minio/minio-go/blob/master/examples/s3/setbucketlifecycle.go

### Expected Result
200 OK and the lifecycle set on the bucket.

### Actual Result

204 No Content 
The client sends an empty body to the server.

### Possible Explanation
If I look at https://github.com/minio/minio-go/blob/a5786a9c2a5b61012d698b18d6ecf308337e8d86/api-put-bucket.go#L265 the content is wrapped into an io.Reader which is then read to EOF and thus has no content left to be read when sent to server over HTTP later on:

https://github.com/minio/minio-go/blob/a5786a9c2a5b61012d698b18d6ecf308337e8d86/api-put-bucket.go#L273

Basically it's doing this: https://play.golang.org/p/T4hrRPXXUKK

Happy to try to fix it and write a test if this really is a bug, but wanted to confirm first.

:)

**Comment by vadmeste** [comment]
@hoempf yes, according to the specifiation, 200 OK should be returned

https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketLifecycleConfiguration.html

Feel free to send a PR.

**Comment by hoempf** [comment]
Hmm, this should reset the string reader to its starting position though:

https://github.com/hoempf/minio-go/blob/a5786a9c2a5b61012d698b18d6ecf308337e8d86/api.go#L607

I must be doing something wrong then ... hmm.

**Comment by vadmeste** [comment]
@hoempf Isn't this a MinIO server issue ?

Could you write a snippet code which shows the issue ?

**Comment by hoempf** [comment]
It could be a server issue.

Maybe I get the 204 because I‘ve already PUT the policy before. But the client returns this as an error (err != nil). Maybe I‘m just expecting different behaviour and everything is working as intended ;)

I‘ll try to come up with reproducible code later today.

**Comment by kannappanr** [comment]
@hoempf We have sent a fix in `MinIO` and now you should see the server returning `200` if you try the example against `MinIO` running latest master. Closing this issue as fixed. Please feel free to reach out to us if you have further questions

**Comment by hoempf** [comment]
Awesome, that fixed it :) Thanks!
